lease/lessor.go

```diff
@@ -160,9 +164,10 @@ func newLessor(b backend.Backend, minLeaseTTL int64) *lessor {
 		b:           b,
 		minLeaseTTL: minLeaseTTL,
 		// expiredC is a small buffered chan to avoid unnecessary blocking.
-		expiredC: make(chan []*Lease, 16),
-		stopC:    make(chan struct{}),
-		doneC:    make(chan struct{}),
+		expiredC:  make(chan []*Lease, 16),
+		stopC:     make(chan struct{}),
+		doneC:     make(chan struct{}),
+		promoting: true,
```

Comment:
Yeah we only call `Promote` after leader election https://github.com/coreos/etcd/blob/master/etcdserver/server.go#L1307-L1319. So should be safe to randomize for every `Promote` call, and no need `promoting` field?